signer/src/utils.rs

```diff
@@ -17,10 +17,19 @@
 macro_rules! once_static {
     ($($(#[$attr:meta])* $vis:vis fn $name:ident() -> $ty:ty { $expr:expr } )+) => {
         $(
-            $(#[$attr])*
-            $vis fn $name() -> &'static $ty {
-                static VAR: std::sync::OnceLock<$ty> = std::sync::OnceLock::new();
-                VAR.get_or_init(|| { $expr })
+            cfg_if::cfg_if! {
+                if #[cfg(feature = "std")] {
+                    $(#[$attr])*
+                    $vis fn $name() -> &'static $ty {
+                        static VAR: std::sync::OnceLock<$ty> = std::sync::OnceLock::new();
+                        VAR.get_or_init(|| { $expr })
+                    }
+                } else {
+                    $(#[$attr])*
+                    $vis fn $name() -> $ty {
```

Comment:
I wonder whether there is a way to have this also be `&'static $ty` (ideally enabling "std" wouldn't change the type signature and thus risk breaking anything).

Offhand I can't think of a good solution to this (we went with this approach in the first place to avoid needing `OnceCell` and `critical-sections` which was annoying per-platform feature flags etc).

Let's just emit nothing here in no-std, ie just don't provide dev keys. This would mean that "std" feature was purely additive which is nice.

Later, we could have this expose two versions of each function; one that is like here and just computes the thing the whole time (eg `$name_new()`), and one that returns `&'static $ty` (eg `$name()`) and is only available on "std". But for that we'd need to be able to concat tokens to build the names which requires another dep like `paste`. We can leave this for another time I think.

Reply:
Oh actually ignore this; `once_static` it only used for acouple of internal funcs, so I think we are ok actually!

I still feel uneasy about the different type signatures being hidden in the macro though so perhaps we remove the magic here, and in the one place its used we just duplicate the fns or something?